### PR TITLE
Static arrays

### PR DESCRIPTION
**Pull request by kanav99**
@ChrisRackauckas

### CODE REVIEW

File: src/AutoOptimize.jl [hunk 1]
```diff
@@ -31,9 +31,23 @@ function auto_optimize(prob::ODEProblem,alg=nothing;
...
 
+      if static
+            verbose && println("Try Converting Arrays to Static Arrays")
+            if prob.u0 isa StaticArray
+                  verbose && println("u0 is already Static Array")
+            else
+                  if N > 10
+                        verbose && println("Size more that 10")
+                  else
+                        prob = remake(prob, u0=MArray{Tuple{size(u0)...}}(u0))
```
Comment (ChrisRackauckas) [inline src/AutoOptimize.jl]: MArray doesn't have the same advantages IIRC. We should modelingtoolkitize and try the OOP version with SA

**Comment by kanav99** [comment]
I would need a problem to check if this works, do you have a problem in mind?

File: src/AutoOptimize.jl [hunk 2]
```diff
@@ -47,13 +49,23 @@ function auto_optimize(prob::ODEProblem,alg=nothing;
...
-                                    sparse = N > SPARSE_CUTOFF &&
-                                             sparsity_percentage < SPARSE_PERCENTAGE_CUTOFF,
-                                    parallel = form,
-                                    prob.kwargs...)
+                  static = static && N < STATIC_ARRAY_CUTOFF
+
+                  if static
+                        prob = ODEProblem{false}(sys,SArray{Tuple{size(prob.u0)...}}(prob.u0),prob.tspan,prob.p,
+                                          jac = true, tgrad = true, simplify = true,
+                                          sparse = N > SPARSE_CUTOFF &&
```
Comment (ChrisRackauckas) [inline src/AutoOptimize.jl]: SA should never be sparse.

File: src/AutoOptimize.jl [hunk 3]
```diff
@@ -47,13 +49,23 @@ function auto_optimize(prob::ODEProblem,alg=nothing;
...
+                                                   sparsity_percentage < SPARSE_PERCENTAGE_CUTOFF,
+                                          parallel = form,
```
Comment (ChrisRackauckas) [inline src/AutoOptimize.jl]: SA is never parallel

**Comment by kanav99** [comment]
I will work on finding the global best setup in the `hyper_auto_optimize`

**Pull request merged**